### ISSUE TITLE
Add profiler to moose containers

### ISSUE THREAD

**Issue by grmnptr** [idaholab/moose#28583]
### Motivation

To get a better picture on the bottlenecks in moose, we should be able to easily profile the code.

### Design

Add gperftools and pprof to the default moose containers.

### Impact

Easier development.

**Comment by grmnptr** [comment]
Yeah I tried to use the conda package today on my linux box with the latest moose dev and it didn't work (couldn't print the source code). Other functionalities seemed to work. So I thought: wouldn't it be cool if I could just load moose-dev on rod and profile my code? Also this would be useful on HPC too, so we can profile large cases.

**Comment by lindsayad** [comment]
Nice
